tasks/main.yml

```diff
@@ -1,2 +1,172 @@
+    name: "{{ services.get(item).vars.aux_pkg|default(default_aux_pkg) }}"
+    state: "{{ services.get(item).vars.aux_pkg_state|default('latest') }}"
+  loop: "{{ pacifica_enabled_services }}"
+  tags:
+    - pacifica
+    - pacifica_python
+
+- name: Install module packages
+  pip:
+    chdir: "{{ services.get(item).vars.virtualenv_path|default('/opt') }}"
+    virtualenv: "{{ services.get(item).vars.virtualenv|default('default') }}"
+    virtualenv_python: "{{ services.get(item).vars.virtualenv_python|default('python2.7') }}"
+    name: "pacifica-{{ services.get(item).vars.pacifica_module }}{{ services.get(item).vars.version|default('') }}"
+    state: "{{ services.get(item).vars.pacifica_module_state|default('latest') }}"
+  notify:
+    - "Pacifica: Reload uWSGI Service"
+  loop: "{{ pacifica_enabled_services }}"
+  tags:
+    - pacifica
+    - pacifica_python
+
+- name: Deploy Pacifica module INI configuration(s)
+  vars:
+    service: "{{ item }}"
+    value: "{{ services.get(item).vars.ini_config|default({}) }}"
+  template:
+    src: templates/config.ini.j2
+    dest: >-
+      {{ services.get(item).vars.virtualenv_path|default('/opt') }}/{{
+      services.get(item).vars.virtualenv|default('default') }}/{{
+      item }}.ini
+  notify:
+    - "Pacifica: Reload uWSGI Service"
+  loop: "{{ pacifica_enabled_services }}"
+  tags:
+    - pacifica
+    - pacifica_python
+
+- name: Deploy Pacifica module CherryPy configuration(s)
+  vars:
+    service: "{{ item }}"
+    value: "{{ default_cp_config | combine(services.get(item).vars.cp_config|default({})) }}"
+  template:
+    src: templates/config-cp.ini.j2
+    dest: >-
+      {{ services.get(item).vars.virtualenv_path|default('/opt') }}/{{
+      services.get(item).vars.virtualenv|default('default') }}/{{
+      item }}-cp.ini
+  notify:
+    - "Pacifica: Reload uWSGI Service"
+  loop: "{{ pacifica_enabled_services }}"
+  tags:
+    - pacifica
+    - pacifica_python
+
+- name: Deploy module service start script(s)
+  register: restart
+  no_log: "{{ enable_no_log|default(True) }}"
+  vars:
+    service: "{{ item }}"
+    value: "{{ services.get(item).vars }}"
+  template:
+    src: templates/start_script.j2
+    dest: >-
+      {{ services.get(item).vars.virtualenv_path|default('/opt') }}/{{
+      services.get(item).vars.virtualenv|default('default') }}/{{
+      item }}
+    owner: root
+    group: root
+    mode: 0750
+  notify:
+    - "Pacifica: Restart uWSGI Service"
+  loop: "{{ pacifica_enabled_services }}"
+  tags:
+    - pacifica
+    - pacifica_python
+
+- name: Deploy systemd unit file(s)
+  vars:
+    service: "{{ item }}"
+    value: "{{ services.get(item).vars }}"
+  template:
+    src: templates/service.j2
+    dest: "/etc/systemd/system/{{ item }}.service"
+    owner: root
+    group: root
+    mode: 0644
+  notify:
+    - "Pacifica: Restart uWSGI Service"
+  loop: "{{ pacifica_enabled_services }}"
+  tags:
+    - pacifica
+    - pacifica_python
```

Comment:
We need to add one more task that says the uWSGI service is supposed to be `started`. We can put a `when` conditional on it in case someone would like to not enforce this on every run but as a minimal procedural task it should be enforced by default.

Reply:
Done